homeassistant/components/lastfm/config_flow.py

```diff
@@ -0,0 +1,66 @@
+"""Config flow for LastFm."""
+from typing import Any
+
+from pylast import LastFMNetwork, WSError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_API_KEY
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.selector import SelectSelector, SelectSelectorConfig
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_USERS, DOMAIN, LOGGER, PLACEHOLDERS
+
+CONFIG_SCHEMA: vol.Schema = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): str,
+        vol.Required(CONF_USERS): SelectSelector(
+            SelectSelectorConfig(options=[], custom_value=True, multiple=True)
+        ),
+    }
+)
+
+
+class LastFmFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Config flow handler for LastFm."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Initialize user input."""
+        errors = {}
+        if user_input is not None:
+            lastfm_api = LastFMNetwork(api_key=user_input[CONF_API_KEY])
+            final_user_input = user_input.copy()
+            try:
+                for user in final_user_input[CONF_USERS]:
+                    lastfm_api.get_user(user).get_playcount()
+            except WSError as error:
+                LOGGER.error(error)
+                if error.details == "User not found":
+                    errors["base"] = "invalid_account"
+                elif (
+                    error.details
+                    == "Invalid API key - You must be granted a valid key by last.fm"
+                ):
+                    errors["base"] = "invalid_auth"
+                else:
+                    errors["base"] = "unknown"
+            except Exception:  # pylint:disable=broad-except
+                errors["base"] = "unknown"
+            if not errors:
+                return self.async_create_entry(title="LastFM", data=final_user_input)
```

Comment:
Should we consider putting it in options instead and implement options flow so the user can edit the list easily?

Reply:
Oh right, forgot to add the OptionsFlow